src/modelapi.ts

```diff
@@ -111,7 +111,7 @@ class ModelApi {
   ) => {
     const subGraph = new SGraph(new SElement('', undefined));
     subGraph.createOrGetElementWithNew(sourceElement);
-    const stack = [sourceElement];
+    const stack = new Set<SElement>([sourceElement]);
```

Comment:
Why the sourceElement is in square brackets?

Reply:
It is because a set needs to be initialized with an iterable (so thus the array).